waku/node/peer_manager/peer_store/waku_peer_storage.nim

```diff
@@ -58,113 +70,104 @@ proc encode*(remotePeerInfo: RemotePeerInfo): PeerStorageResult[ProtoBuffer] =
+    (seq[byte], seq[byte]),
+    void
+  ).expect("Valid statement")
 
-  ## General initialization
+  # General initialization
+  let ps = WakuPeerStorage(database: db, replaceStmt: replaceStmt)
 
-  ok(WakuPeerStorage(database: db,
-                     replaceStmt: replaceStmt))
-
-
-method put*(db: WakuPeerStorage,
-            peerId: PeerID,
-            remotePeerInfo: RemotePeerInfo,
-            connectedness: Connectedness,
-            disconnectTime: int64): PeerStorageResult[void] =
+  return ok(ps)
 
+method put*(
+  db: WakuPeerStorage,
+  remotePeerInfo: RemotePeerInfo
+  ): PeerStorageResult[void] =
   ## Adds a peer to storage or replaces existing entry if it already exists
-  let encoded = remotePeerInfo.encode()
-
-  if encoded.isErr:
-    return err("failed to encode: " & encoded.error())
+  
+  let encoded = remotePeerInfo.encode().valueOr:
+    return err("peer info encoding failed: " & error)
 
-  let res = db.replaceStmt.exec((peerId.data, encoded.get().buffer, int32(ord(connectedness)), disconnectTime))
-  if res.isErr:
-    return err("failed")
+  db.replaceStmt.exec((remotePeerInfo.peerId.data, encoded.buffer)).isOkOr:
+    return err("DB operation failed: " & error)
 
-  ok()
+  return ok()
 
-method getAll*(db: WakuPeerStorage, onData: peer_storage.DataProc): PeerStorageResult[bool] =
+method getAll*(
+  db: WakuPeerStorage,
+  onData: peer_storage.DataProc
+  ): PeerStorageResult[void] =
   ## Retrieves all peers from storage
-  var gotPeers = false
-
-  proc peer(s: ptr sqlite3_stmt) {.raises: [Defect, LPError, ResultError[ProtoError]].} =
-    gotPeers = true
+  
+  proc peer(s: ptr sqlite3_stmt) {.raises: [ResultError[ProtoError]].} =
     let
-      # Peer ID
-      pId = cast[ptr UncheckedArray[byte]](sqlite3_column_blob(s, 0))
-      pIdL = sqlite3_column_bytes(s, 0)
-      peerId = PeerID.init(@(toOpenArray(pId, 0, pIdL - 1))).tryGet()
       # Stored Info
       sTo = cast[ptr UncheckedArray[byte]](sqlite3_column_blob(s, 1))
       sToL = sqlite3_column_bytes(s, 1)
-      storedInfo = RemotePeerInfo.init(@(toOpenArray(sTo, 0, sToL - 1))).tryGet()
-      # Connectedness
-      connectedness = Connectedness(sqlite3_column_int(s, 2))
-      # DisconnectTime
-      disconnectTime = sqlite3_column_int64(s, 3)
+      storedInfo = RemotePeerInfo.decode(@(toOpenArray(sTo, 0, sToL - 1))).tryGet()
+
+    onData(storedInfo)
 
-    onData(peerId, storedInfo, connectedness, disconnectTime)
+  let catchRes = catch: db.database.query("SELECT peerId, storedInfo FROM Peer", peer)
 
-  var queryResult: DatabaseResult[bool]
-  try:
-    queryResult = db.database.query("SELECT peerId, storedInfo, connectedness, disconnectTime FROM Peer", peer)
-  except LPError, ResultError[ProtoError]:
-    return err("failed to extract peer from query result")
+  let queryRes =
+    if catchRes.isErr():
+      return err("failed to extract peer from query result: " & catchRes.error.msg)
+    else: catchRes.get()
```

Comment:
We can't use the traditional `isOkOr` because of how the `catch` template works, right?

Reply:
> We can't use the traditional `isOkOr` because of how the `catch` template works, right?

Yeah it should be possible but the compiler not smart enough I would guess.